.github/workflows/pin-requirements.yml

```
@@ -0,0 +1,48 @@
+name: Pin Requirements
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  update-pinned-requirements:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+        with:
+          ref: main
+
+      - uses: actions/setup-python@d27e3f3d7c64b4bbf8e4abfb9b63b83e846e0435
+        with:
+          python-version: "3.x"
+          cache: "pip"
+          cache-dependency-path: pyproject.toml
+
+      - run: pip install pip-tools
+
+      - name: Determine latest sigstore-python PyPI release
+        run: |
+          version=$(curl -H "Accept: application/vnd.pypi.simple.v1+json" https://pypi.org/simple/sigstore/ | jq -r '.versions[-1]')
+          echo "sigstore_python_version=$version" >>$GITHUB_ENV
+      - name: Update requirements
+        run: |
+          cd install
+          echo "sigstore==${{ env.sigstore_python_version }}" >requirements.in
+          pip-compile --allow-unsafe --generate-hashes --output-file=requirements.txt requirements.in
+      - name: Open pull request
+        id: pr
+        uses: peter-evans/create-pull-request@2b011faafdcbc9ceb11414d64d0573f37c774b04 # v4.2.3
+        with:
+          title: Update pinned requirements for v${{ env.sigstore_python_version }}
+          body: Pins dependencies for [v${{ env.sigstore_python_version }}](https://github.com/sigstore/sigstore-python/releases/tag/v${{ env.sigstore_python_version }}).
+          commit-message: "install: update pinned requirements"
```

Just for visual scanning:

```suggestion
          commit-message: "[BOT] install: update pinned requirements"
```